clippy_lints/src/try_err.rs

```diff
@@ -92,7 +92,7 @@ impl<'tcx> LateLintPass<'tcx> for TryErr {
 
                 let expr_err_ty = cx.typeck_results().expr_ty(err_arg);
 
-                let origin_snippet = if err_arg.span.from_expansion() {
+                let origin_snippet = if err_arg.span.from_expansion() && !in_macro(expr.span) {
```

Comment:
This change seems unrelated?

Reply:
This `!in_macro(expr.span)` is here to prevent the use of the macro callsite for suggestion.
In this example:
```
macro_rules! try_validation {
    ($e: expr) => {{
        match $e {
            Ok(_) => 0,
            Err(_) => Err(1)?, // <-- lint here
        }
    }};
}

fn calling_macro() -> Result<i32, i32> {
    try_validation!(Ok::<_, i32>(5));
    Ok(5)
}
```
For information,`err_arg.span` is referring to the `1` from `Err(1)?`
We want the suggestion to be `return Err(1)` and not to be the macro call site (`try_validation!(Ok::<_, i32>(5));`).